element-connector/src/test/java/org/eclipse/californium/elements/util/CertPathUtilTest.java

```diff
@@ -169,117 +206,327 @@ public void testServerCertificateValidationWithoutTrust() throws Exception {
-		CertPath verifiedPath = CertPathUtil.validateCertificatePath(false, certPath, trusts);
-		assertEquals(Arrays.asList(certificates), verifiedPath.getCertificates());
+		List<X509Certificate> certificates = TestCertificatesTools.getServerCertificateChainAsList();
+		X509Certificate[] trusts = new X509Certificate[] { TestCertificatesTools.getTrustedCA() };
+		CertPath certPath = CertPathUtil.generateCertPath(certificates);
+		CertPathUtil.validateCertificatePathWithIssuer(false, certPath, trusts);
 	}
 
+	/**
+	 * Certificate-path: server, ca
+	 * Trust: "first match", ca
+	 * Expected result: pass => server ca
+	 * @throws Exception if an unexpected error occurs
+	 */
 	@Test
 	public void testServerCertificateTruncatingValidationWithIntermediateTrust() throws Exception {
-		X509Certificate[] certificates = TestCertificatesTools.getServerCertificateChain();
-		X509Certificate[] trusts = new X509Certificate[] { certificates[1] };
-		X509Certificate[] verfied = new X509Certificate[] { certificates[0] };
-		CertPath certPath = CertPathUtil.generateCertPath(Arrays.asList(certificates));
-		CertPath verifiedPath = CertPathUtil.validateCertificatePath(true, certPath, trusts);
-		assertEquals(Arrays.asList(verfied), verifiedPath.getCertificates());
+		List<X509Certificate> certificates = TestCertificatesTools.getServerCertificateChainAsList();
+		X509Certificate[] trusts = new X509Certificate[] { TestCertificatesTools.getTrustedCA() };
+		CertPath certPath = CertPathUtil.generateCertPath(certificates);
+		CertPath verifiedPath = CertPathUtil.validateCertificatePathWithIssuer(true, certPath, trusts);
+		TestCertificatesTools.assertEquals(certificates, verifiedPath.getCertificates());
+	}
+
+	/**
+	 * Certificate-path: server
+	 * 1.
+	 * Trust: ca, caalt
+	 * Expected result: pass => server, ca
+	 * 2.
+	 * Trust: caalt, ca
+	 * Expected result: pass => server, ca
+	 * @throws Exception if an unexpected error occurs
+	 */
+	@Test
+	public void testServerCertificateTruncatingValidationWithAmbiguousTrust() throws Exception {
+		X509Certificate server = TestCertificatesTools.getServerCertificateChain()[0];
+		X509Certificate ca = TestCertificatesTools.getTrustedCA();
+		X509Certificate caalt = TestCertificatesTools.getAlternativeCA();
+		X509Certificate[] path = new X509Certificate[] { server };
+		X509Certificate[] trusts = new X509Certificate[] { ca, caalt };
+		X509Certificate[] verfied = new X509Certificate[] { server, ca };
+		CertPath certPath = CertPathUtil.generateCertPath(Arrays.asList(path));
+		CertPath verifiedPath = CertPathUtil.validateCertificatePathWithIssuer(false, certPath, trusts);
+		TestCertificatesTools.assertEquals(verfied, verifiedPath.getCertificates());
+
+		X509Certificate[] trusts2 = new X509Certificate[] { caalt, ca };
+		verifiedPath = CertPathUtil.validateCertificatePathWithIssuer(false, certPath, trusts2);
+		TestCertificatesTools.assertEquals(verfied, verifiedPath.getCertificates());
```

Comment:
Should it be : 
```java
CertPath verifiedPath = CertPathUtil.validateCertificatePathWithIssuer(true, certPath, trusts); 
```
?

Reply:
The test-names are not that "self explaining" .... 

Truncating: the send certificate path is truncated
Truncated: the path may be truncated on validation.

I will add the missing test-case.

Reply:
I pushed the additional test case to master.